provider/macros/src/lib.rs

```diff
@@ -92,12 +92,167 @@ mod tests;
+                        if name == "fallback_by" {
+                            at_most_one_option(
+                                &mut fallback_by,
+                                value,
+                                "fallback_by",
+                                input.span(),
+                            )?;
+                        } else if name == "extension_key" {
+                            at_most_one_option(
+                                &mut extension_key,
+                                value,
+                                "extension_key",
+                                input.span(),
+                            )?;
+                        } else if name == "fallback_supplement" {
+                            at_most_one_option(
+                                &mut fallback_supplement,
+                                value,
+                                "fallback_supplement",
+                                input.span(),
+                            )?;
+                        }
+                    }
+                    DataStructMarkerArg::Lit(lit) => {
+                        at_most_one_option(&mut key_lit, lit, "literal key", input.span())?;
+                    }
+                }
+            }
+            let marker_name = if let Some(marker_name) = marker_name {
+                marker_name
+            } else {
+                return Err(parse::Error::new(
+                    input.span(),
+                    "marker() must contain a marker!",
+                ));
+            };
+
+            Ok(Self {
+                marker_name,
+                key_lit,
+                fallback_by,
+                extension_key,
+                fallback_supplement,
+            })
+        } else {
+            let mut this = DataStructArg::new(path);
+            let lookahead = input.lookahead1();
+            if lookahead.peek(Token![=]) {
+                let _t: Token![=] = input.parse()?;
+                let lit: LitStr = input.parse()?;
+                this.key_lit = Some(lit);
+                Ok(this)
+            } else {
+                Ok(this)
+            }
+        }
+    }
+}
+
+/// A single argument to `marker()` in `#[data_struct(..., marker(...), ...)]
+enum DataStructMarkerArg {
+    Path(Path),
+    NameValue(Ident, LitStr),
+    Lit(LitStr),
+}
+impl Parse for DataStructMarkerArg {
+    fn parse(input: ParseStream<'_>) -> parse::Result<Self> {
+        let lookahead = input.lookahead1();
+        if lookahead.peek(LitStr) {
+            Ok(DataStructMarkerArg::Lit(input.parse()?))
+        } else {
+            let path: Path = input.parse()?;
+            let lookahead = input.lookahead1();
+            if lookahead.peek(Token![=]) {
+                let _tok: Token![=] = input.parse()?;
+                let ident = path.get_ident().ok_or_else(|| {
+                    parse::Error::new(input.span(), "Expected identifier before `=`, found path")
```

Comment:
Thought: I think `input.span()` is pointing at the token after the `=`, but the problematic one is the span before the `=`